### PR TITLE
feat: baseline workflow

### PR DESCRIPTION
**Pull request by mohamed-cds**
Initial workflow files for terragrunt. The goal is to use the `satellite_accounts.json` file to retrieve the list of accounts and dynamically create the github action workflow matrix.

For example if account `028051698106` has the minimum IAM permissions bootstraped, to have it included in the workflow you'd update the `satellite_accounts.json` as follows:

```json
[
  {"account":"028051698106"}
]
```

The goal is to also use this JSON in our terragrunt environment so that cbs supported satellite accounts only reside in one place.

This results in a transformation being required in the Github action workflow so that the JSON file follows the pattern `{"include": [{"account":"028051698106"}]}` so that it can be used directly in matrix steps of the github workflow.

### CODE REVIEW

File: .github/workflows/terragrunt-apply-satellite.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      - uses: cds-snc/paths-filter@v2.10.3
+        id: filter
+        with:
+          filters: |
+            config:
+              - 'terragrunt/aws/config/**' 
+              - 'terragrunt/env/satellite/config/**'             
+            s3:
+              - 'terragrunt/aws/s3/**'  
+              - 'terragrunt/env/satellite/s3/**'
```
Comment (patheard) [inline .github/workflows/terragrunt-apply-satellite.yml:84]: This got unhelpfully renamed by me to `satellite_bucket` in the folder structure.

File: .github/workflows/terragrunt-plan-satellite.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Read account file
+        id: set-matrix
+        run: |
+          accounts=$(jq -n -c "{\"include\": [inputs | {"account": .} ] }"  satellite_accounts.json)
```
Comment (patheard) [inline .github/workflows/terragrunt-plan-satellite.yml:36]: Ah `jq` the most magic of command line tools 🪄

File: .github/workflows/terragrunt-apply-satellite.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Read account file
+        id: set-matrix
+        run: |
+          accounts=$(jq -c "{\"include\": . }" satellite_accounts.json)
```
Comment (patheard) [inline .github/workflows/terragrunt-apply-satellite.yml:38]: Should this be the same `jq` command that's being used in the `terragrunt-plan-satellite.yml` workflow?

File: terragrunt/aws/central_account/inputs.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
+variable "billing_code" {
+  description = "The billing code to tag our resources with"
+  type        = string
+}
```
Comment (patheard) [inline terragrunt/aws/central_account/inputs.tf:4]: This is getting handled by the global `billing_tag_value` we set in the root `terragrunt.hcl`.

**Review by patheard — COMMENTED** [review]
Had a few questions/comments for you

File: .github/workflows/terragrunt-apply-satellite.yml (hunk 3 above)
Comment (mohamed-cds) [reply .github/workflows/terragrunt-apply-satellite.yml]: 💯

**Pull request merged**